### PR TITLE
Tickets/dm 24423

### PR DESCRIPTION
**Pull request by teweitsai**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ pipeline {
         // It is recommended by SQUARE team do not add the label and let the
         // sytem decide.
         docker {
-            image 'lsstts/aos:w_2020_14'
+            image 'lsstts/aos:w_2020_15'
```
Comment (pkubanek) [inline Jenkinsfile:11]: Should this work? It's much better to define just once.. You can still change SIMS_VERSION independent of the docker version if needed.

```patch
diff --git a/Jenkinsfile b/Jenkinsfile
index 060f30b..f603c59 100644
--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -2,13 +2,26 @@

 pipeline {

+    environment {
+        // Development tool set
+        DEV_TOOL="/opt/rh/devtoolset-8/enable"
+        // Position of LSST stack directory
+        LSST_STACK="/opt/lsst/software/stack"
+        // Pipeline Sims Version
+        SIMS_VERSION="sims_w_2020_15"
+        // XML report path
+        XML_REPORT="jenkinsReport/report.xml"
+        // Module name used in the pytest coverage analysis
+        MODULE_NAME="lsst.ts.ofc"
+    }
+
     agent {
         // Use the docker to assign the Python version.
         // Use the label to assign the node to run the test.
         // It is recommended by SQUARE team do not add the label and let the
         // sytem decide.
         docker {
-            image 'lsstts/aos:w_2020_15'
+            image "lsstts/${env.SIMS_VERSION}"
             args '-u root'
         }
     }
@@ -17,18 +30,6 @@ pipeline {
         pollSCM('H * * * *')
     }

-    environment {
-        // Development tool set
-        DEV_TOOL="/opt/rh/devtoolset-8/enable"
-        // Position of LSST stack directory
-        LSST_STACK="/opt/lsst/software/stack"
-        // Pipeline Sims Version
-        SIMS_VERSION="sims_w_2020_15"
-        // XML report path
-        XML_REPORT="jenkinsReport/report.xml"
-        // Module name used in the pytest coverage analysis
-        MODULE_NAME="lsst.ts.ofc"
-    }

     stages {
         stage ('Install Requirements') {
```

File: python/lsst/ts/ofc/OptCtrlDataDecorator.py [hunk 2]
```diff
@@ -162,35 +155,32 @@ def _getRbStroke(self):
...
-        mat = paramReader.getMatContent()
-        bendingMode = mat[:, usecols]
+        bendingModeToForce = BendModeToForce()
+        bendingModeToForce.config(self.configDir, dofGroup,
+                                  actuatorForceFileName)
+        bendingMode = bendingModeToForce.getRotMat()
+
         authority = np.std(bendingMode, axis=0)
 
         return authority
```
Comment (pkubanek) [inline python/lsst/ts/ofc/OptCtrlDataDecorator.py:186]: if you change this, consider change to "return np.std(..)!" as well. See above for patch in Iter...

File: python/lsst/ts/ofc/SubSysAdap.py [hunk 3]
```diff
@@ -73,10 +69,9 @@ def _getRotMatMirror(self, configDir, dofGroup, rotMatfileName):
         rotMatFilePath = os.path.join(configDir, mirrorDirName, rotMatfileName)
         rotMatFile = ParamReader(filePath=rotMatFilePath)
 
-        rotMatBend = rotMatFile.getSetting("bend")
         rotMatAct = rotMatFile.getSetting("act")
 
-        return np.diag(rotMatBend), rotMatAct
+        return rotMatAct
```
Comment (pkubanek) [inline python/lsst/ts/ofc/SubSysAdap.py:74]: Rather change to "return rotMatFile.getSetting…."

File: Jenkinsfile (hunk 1 above)
Comment (teweitsai) [reply Jenkinsfile:11]: This method will give the error:
```
/home/jenkins/workspace/_Site_ts_ofc_feature_testJenkins@tmp/durable-94fae833/script.sh: line 1: lsstts/aos:${env.LSST_VERSION}: bad substitution
```
and

```
Error when executing always post condition:
org.jenkinsci.plugins.workflow.steps.MissingContextVariableException: Required context class hudson.FilePath is missing
```
The clear test report is at:
https://tssw-ci.lsst.org/job/LSST_Telescope-and-Site/job/ts_ofc/job/feature%252FtestJenkins/1/console

File: python/lsst/ts/ofc/IterDataReader.py [hunk 4]
```diff
@@ -28,23 +30,63 @@ def setDataDir(self, dataDir):
 
         self.dataDir = dataDir
 
-    def _getIterDir(self, iterNum):
-        """Get the directory of specific iteration data.
+    def getAbsFilePathOfWfsErr(self, iterNum, reMatchStr=r"wfs.zer"):
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply python/lsst/ts/ofc/IterDataReader.py:22]: Fixed in the commit: 8c1ea57. Thanks!

File: python/lsst/ts/ofc/OptCtrlDataDecorator.py (hunk 2 above)
Comment (teweitsai) [reply python/lsst/ts/ofc/OptCtrlDataDecorator.py]: Fixed in the commit: a2945e0. Thanks!

File: python/lsst/ts/ofc/SubSysAdap.py (hunk 3 above)
Comment (teweitsai) [reply python/lsst/ts/ofc/SubSysAdap.py]: Fixed in the commit: a2945e0. Thanks!

**Pull request merged**